src/gauss_legendre.jl

```diff
@@ -0,0 +1,1290 @@
+        @. wgts[imin[j] - k + 1:imax[j]] += element_scale[j]*w[1:ngrid] 
+        
+        k = 2        
+    end
+    return grid, wgts
+end
+
+"""
+function for setting up the full Gauss-Legendre-Radau
+grid and collocation point weights
+see comments of Gauss-Legendre-Lobatto routine above
+"""
+function scaled_gauss_legendre_radau_grid(ngrid, nelement_local, n_local, element_scale, element_shift, imin, imax, irank)
+    # get Gauss-Legendre-Lobatto points and weights on [-1,1]
+    x_lob, w_lob = gausslobatto(ngrid)
+    # get Gauss-Legendre-Radau points and weights on [-1,1)
+    x_rad, w_rad = gaussradau(ngrid)
+    # transform to a Gauss-Legendre-Radau grid on (-1,1]
+    x_rad, w_rad = -reverse(x_rad), reverse(w_rad)#
+    # grid and weights arrays
+    grid = allocate_float(n_local)
+    wgts = allocate_float(n_local)
+    wgts .= 0.0
+    if irank == 0
+        # for 1st element, fill in with Gauss-Legendre-Radau points
+        j = 1
+        # element_scale[j]
+        # element_shift[j]
+        # factor with maps [-1,1] -> a subset of [-L/2, L/2]
+        @. grid[imin[j]:imax[j]] = element_scale[j]*x_rad[1:ngrid] + element_shift[j]
+        @. wgts[imin[j]:imax[j]] += element_scale[j]*w_rad[1:ngrid]       
+        #integer to deal with the overlap of element boundaries
+        k = 2
+        @inbounds for j in 2:nelement_local
+            # element_scale[j]
+            # element_shift[j]
+            # factor with maps [-1,1] -> a subset of [-L/2, L/2]
+            @. grid[imin[j]:imax[j]] = element_scale[j]*x_lob[k:ngrid] + element_shift[j]
+            @. wgts[imin[j] - k + 1:imax[j]] += element_scale[j]*w_lob[1:ngrid]         
+        end
+    else # all elements are Gauss-Legendre-Lobatto
+        #integer to deal with the overlap of element boundaries
+        k = 1
+        @inbounds for j in 1:nelement_local
+            # element_scale[j]
+            # element_shift[j]
+            # factor with maps [-1,1] -> a subset of [-L/2, L/2]
+            @. grid[imin[j]:imax[j]] = element_scale[j]*x_lob[k:ngrid] + element_shift[j]
+            @. wgts[imin[j] - k + 1:imax[j]] += element_scale[j]*w_lob[1:ngrid]            
+            k = 2 
+        end
+    end
+    return grid, wgts
+end
+
+"""
+function that assigns the local weak-form matrices to 
+a global array QQ_global for later solving weak form of required
+1D equation
+
+option choosing type of matrix to be constructed -- "M" (mass matrix), "S" (derivative matrix)
+"""
+function setup_global_weak_form_matrix!(QQ_global::Array{mk_float,2},
+                               lobatto::gausslegendre_base_info,
+                               radau::gausslegendre_base_info, 
+                               coord,option)
+    QQ_j = allocate_float(coord.ngrid,coord.ngrid)
+    QQ_jp1 = allocate_float(coord.ngrid,coord.ngrid)
+    
+    ngrid = coord.ngrid
+    imin = coord.imin
+    imax = coord.imax
+    @. QQ_global = 0.0
+    mass_matrix = (option == "M") && false

```

Comment:
At the moment the code is meant to be general enough to solve a second order ODE in 1D. The flags are set at the moment so that no boundary conditions are used, but if there was a reason to solve a 1D ODE in weak form, we would want those branches back. Maybe the code can be made clearer, but I don't think we should delete it outright.

Reply:
@johnomotani Are you happy with my resolution?https://github.com/mabarnes/moment_kinetics/pull/149/commits/97caaad1de5216848f6c5575bc989f0fc26503b3.

Reply:
Looks good to me, thanks!